reactor-netty-core/src/test/java/reactor/netty/transport/ProxyProviderTest.java

```diff
@@ -204,6 +207,171 @@ void shouldNotCreateProxyProviderWithMissingRemoteHostInfo() {
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getAddress().get().getPort()).isEqualTo(8443);
+	}
+
+	@Test
+	void proxyFromSystemProperties_defaultNonHttpHostsSetForHttpProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTP_PROXY_HOST, "host");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getNonProxyHostsPredicate().toString()).isEqualTo(ProxyProvider.DEFAULT_NON_PROXY_HOSTS);
+	}
+
+	@Test
+	void proxyFromSystemProperties_defaultNonHttpHostsSetForHttpsProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTPS_PROXY_HOST, "host");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getNonProxyHostsPredicate().toString()).isEqualTo(ProxyProvider.DEFAULT_NON_PROXY_HOSTS);
+	}
+
+	@Test
+	void proxyFromSystemProperties_customNonProxyHostsSetForHttpProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTP_PROXY_HOST, "host");
+		properties.setProperty(ProxyProvider.HTTP_NON_PROXY_HOSTS, "non-host");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getNonProxyHostsPredicate().toString()).isEqualTo("non-host");
+	}
+
+	@Test
+	void proxyFromSystemProperties_customNonProxyHostsSetForHttpsProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTPS_PROXY_HOST, "host");
+		properties.setProperty(ProxyProvider.HTTP_NON_PROXY_HOSTS, "non-host");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getNonProxyHostsPredicate().toString()).isEqualTo("non-host");
+	}
+
+	@Test
+	void proxyFromSystemProperties_socksProxy5SetWhenSocksSystemPropertySet() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.SOCKS_PROXY_HOST, "host");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getType()).isEqualTo(ProxyProvider.Proxy.SOCKS5);
+		assertThat(provider.getAddress().get().getHostName()).isEqualTo("host");
+	}
+
+	@Test
+	void proxyFromSystemProperties_overrideSocksVersionWithCustomProperty() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.SOCKS_PROXY_HOST, "host");
+		properties.setProperty(ProxyProvider.SOCKS_VERSION, "4");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getType()).isEqualTo(ProxyProvider.Proxy.SOCKS4);
+	}
+
+	@Test
+	void proxyFromSystemProperties_overrideSocksPortWithCustomProperty() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.SOCKS_PROXY_HOST, "host");
+		properties.setProperty(ProxyProvider.SOCKS_PROXY_PORT, "2080");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getAddress().get().getPort()).isEqualTo(2080);
```

Comment:
Add assertion that `provider` is not null.

Reply:
done